src/regress/lib/libcrypto/md/md_test.c

```diff
@@ -0,0 +1,302 @@
+			0x31, 0xd6, 0xcf, 0xe0, 0xd1, 0x6a, 0xe9, 0x31,
+			0xb7, 0x3c, 0x59, 0xd7, 0xe0, 0xc0, 0x89, 0xc0,
+		}
+	},
+	{
+		.algorithm = NID_md4,
+		.in = "a",
+		.in_len = 1,
+		.out = {
+			0xbd, 0xe5, 0x2c, 0xb3, 0x1d, 0xe3, 0x3e, 0x46,
+			0x24, 0x5e, 0x05, 0xfb, 0xdb, 0xd6, 0xfb, 0x24,
+		}
+	},
+	{
+		.algorithm = NID_md4,
+		.in = "abc",
+		.in_len = 3,
+		.out = {
+			0xa4, 0x48, 0x01, 0x7a, 0xaf, 0x21, 0xd8, 0x52,
+			0x5f, 0xc1, 0x0a, 0xe8, 0x7a, 0xa6, 0x72, 0x9d,
+		}
+	},
+	{
+		.algorithm = NID_md4,
+		.in = "message digest",
+		.in_len = 14,
+		.out = {
+			0xd9, 0x13, 0x0a, 0x81, 0x64, 0x54, 0x9f, 0xe8,
+			0x18, 0x87, 0x48, 0x06, 0xe1, 0xc7, 0x01, 0x4b,
+		}
+	},
+	{
+		.algorithm = NID_md4,
+		.in = "abcdefghijklmnopqrstuvwxyz",
+		.in_len = 26,
+		.out = {
+			0xd7, 0x9e, 0x1c, 0x30, 0x8a, 0xa5, 0xbb, 0xcd,
+			0xee, 0xa8, 0xed, 0x63, 0xdf, 0x41, 0x2d, 0xa9,
+		}
+	},
+	{
+		.algorithm = NID_md4,
+		.in =
+		    "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuv"
+		    "wxyz0123456789",
+		.in_len = 62,
+		.out = {
+			0x04, 0x3f, 0x85, 0x82, 0xf2, 0x41, 0xdb, 0x35,
+			0x1c, 0xe6, 0x27, 0xe1, 0x53, 0xe7, 0xf0, 0xe4,
+		}
+	},
+	{
+		.algorithm = NID_md4,
+		.in =
+		    "123456789012345678901234567890123456789012345678"
+		    "90123456789012345678901234567890",
+		.in_len = 80,
+		.out = {
+			0xe3, 0x3b, 0x4d, 0xdc, 0x9c, 0x38, 0xf2, 0x19,
+			0x9c, 0x3e, 0x7b, 0x16, 0x4f, 0xcc, 0x05, 0x36,
+		}
+	},
+
+	/* MD5 (RFC 1321 test vectors)  */
+	{
+		.algorithm = NID_md5,
+		.in = "",
+		.in_len = 0,
+		.out = {
+			0xd4, 0x1d, 0x8c, 0xd9, 0x8f, 0x00, 0xb2, 0x04,
+			0xe9, 0x80, 0x09, 0x98, 0xec, 0xf8, 0x42, 0x7e,
+		}
+	},
+	{
+		.algorithm = NID_md5,
+		.in = "a",
+		.in_len = 1,
+		.out = {
+			0x0c, 0xc1, 0x75, 0xb9, 0xc0, 0xf1, 0xb6, 0xa8,
+			0x31, 0xc3, 0x99, 0xe2, 0x69, 0x77, 0x26, 0x61,
+		}
+	},
+	{
+		.algorithm = NID_md5,
+		.in = "abc",
+		.in_len = 3,
+		.out = {
+			0x90, 0x01, 0x50, 0x98, 0x3c, 0xd2, 0x4f, 0xb0,
+			0xd6, 0x96, 0x3f, 0x7d, 0x28, 0xe1, 0x7f, 0x72,
+		}
+	},
+	{
+		.algorithm = NID_md5,
+		.in = "message digest",
+		.in_len = 14,
+		.out = {
+			0xf9, 0x6b, 0x69, 0x7d, 0x7c, 0xb7, 0x93, 0x8d,
+			0x52, 0x5a, 0x2f, 0x31, 0xaa, 0xf1, 0x61, 0xd0,
+		}
+	},
+	{
+		.algorithm = NID_md5,
+		.in = "abcdefghijklmnopqrstuvwxyz",
+		.in_len = 26,
+		.out = {
+			0xc3, 0xfc, 0xd3, 0xd7, 0x61, 0x92, 0xe4, 0x00,
+			0x7d, 0xfb, 0x49, 0x6c, 0xca, 0x67, 0xe1, 0x3b,
+		}
+	},
+	{
+		.algorithm = NID_md5,
+		.in =
+		    "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuv"
+		    "wxyz0123456789",
+		.in_len = 62,
+		.out = {
+			0xd1, 0x74, 0xab, 0x98, 0xd2, 0x77, 0xd9, 0xf5,
+			0xa5, 0x61, 0x1c, 0x2c, 0x9f, 0x41, 0x9d, 0x9f,
+		}
+	},
+	{
+		.algorithm = NID_md5,
+		.in =
+		    "123456789012345678901234567890123456789012345678"
+		    "90123456789012345678901234567890",
+		.in_len = 80,
+		.out = {
+			0x57, 0xed, 0xf4, 0xa2, 0x2b, 0xe3, 0xc9, 0x55,
+			0xac, 0x49, 0xda, 0x2e, 0x21, 0x07, 0xb6, 0x7a,
+		}
+	}
```

Comment:
Better to leave a trailing comma to avoid noise in future diffs that extend the tests.